### ISSUE TITLE
Functions cannot be deployed failing with "ensure that the code is valid"

### ISSUE THREAD

**Issue by wokalski** [graphcool/prisma#1512]
**Current behavior**

After creating a project and deploying it locally, I've played with it a bit, added some functions and everything was fine. I modified them once or twice. However, the next time I modified them I got `ensure that the code is valid` whatever I did. Even if the body of the function was as simple as it could be. (Like `f (x) { return x }` for operationBefore).

I even removed all functions and left only the ones that have not been previously modified but the error persisted.

The bottom line is that it seems that it's quite easy (but not obvious how) to cause some invalid internal state which causes this issue. After removing all docker images and data connected to them, I could successfully deploy the functions.

**Reproduction**

I'm very sorry - I don't have a reproducible case but I did my best to describe the issue.

**Expected behavior?**

The functions should just be created/updated

**Comment by lastmjs** [comment]
I'm having similar issues. As I try to deploy I always get "ensure that the code is valid".

**Comment by lastmjs** [comment]
It seems like if I get rid of the default export, then it works. But without the export I'm not sure how Graphcool is going to call my code.

**Comment by marktani** [comment]
Does this occur when actually deploying, or when doing a dry run deployment?

**Comment by wokalski** [comment]
When actually locally deploying.

**Comment by lastmjs** [comment]
@marktani For me it is during a dry run

**Comment by wokalski** [comment]
I haven't had this error since upgrading graphcool-framework
